packages/compass-preferences-model/src/storage.ts

```diff
@@ -7,43 +7,11 @@ import {
   getPreferencesValidator,
 } from './preferences-schema';
 import type {
-  AllPreferences,
   StoredPreferences,
   StoredPreferencesValidator,
 } from './preferences-schema';
 
-export interface BasePreferencesStorage {
-  setup(): Promise<void>;
-  getPreferences(): StoredPreferences;
-  updatePreferences(attributes: Partial<StoredPreferences>): Promise<void>;
-}
-
-export class InMemoryStorage implements BasePreferencesStorage {
-  private preferences = getDefaultsForStoredPreferences();
-
-  constructor(preferencesOverrides?: Partial<AllPreferences>) {
-    this.preferences = {
-      ...this.preferences,
-      ...preferencesOverrides,
-    };
-  }
-
-  getPreferences() {
-    return this.preferences;
-  }
-
-  // eslint-disable-next-line @typescript-eslint/require-await
-  async updatePreferences(attributes: Partial<StoredPreferences>) {
-    this.preferences = {
-      ...this.preferences,
-      ...attributes,
-    };
-  }
-
-  async setup() {
-    // noop
-  }
-}
+import type { BasePreferencesStorage } from './in-memory-storage';
```

Comment:
this whole naming and relationship between types looks confusing at first to me. 
looks like you are going for something like:`BasePreferencesStorage` is the main interface that every storages must implement. Then you have 2 implementation, one "environment agnostic" `InMemoryStorage` and one compass-only `PersistentStorage`.

It seems you are avoiding to change the existing persistent code, but i think that if we don't we would model this in an unclear way, making it look like there is a relationship between the in memory storage and the persistent one when there is none.

What about we extract `BasePreferencesStorage` to be outside of `./in-memory-storage` in its own file (and possibly just call it `PreferencesStorage`) and we require this interface in both `./in-memory-storage` and `/src/storage.ts`.

Also since now the class is called PersistentStorage, i think it would be clearer if we make the file name reflect that and rename `/src/storage.ts` to persistent-storage.ts`.

Reply:
Addressed this in my recent commit